common/common.go

```diff
@@ -751,23 +751,28 @@ func ListTriggersInFunc(c *cli.Context, client *fnclient.Fn, fn *modelsv2.Fn) ([
 func DockerRunInitImage(initImage string, fName string) error {
 	fmt.Println("Running init-image: " + initImage)
 
-	// Run the initImage
-	var c1ErrB bytes.Buffer
+	var errB bytes.Buffer
 	tarR, tarW := io.Pipe()
 
-	c1 := exec.Command("docker", "run", "--rm", "-e", "FN_FUNCTION_NAME="+fName, initImage)
-	c1.Stderr = &c1ErrB
-	c1.Stdout = tarW
-
-	c1Err := c1.Start()
-	if c1Err != nil {
-		fmt.Println(c1ErrB.String())
+	cmd := exec.Command("docker", "run", "--rm", "-e",
```

Comment:
it's easier to use CommandContext() here: https://golang.org/pkg/os/exec/#CommandContext

Reply:
Thanks for the pointer - CommandContext looks useful when you want to wrap a Command in a timeout. However, that's not the case here. In the error case, the command *does* return after ~1 minute, but it doesn't push any data onto stdout for the `untarStream` command to consume - thus resulting in `tr.Next()` blocking. Hence wrapping `tr.Next()` in a timeout as opposed to the Command.

Perhaps I'm not understanding your suggestion properly - I'll try to play around with this.

Reply:
Nvm. I think I've figured out a more elegant approach.